src/gui/PluginPinConnectorView.cpp

```diff
@@ -89,9 +89,13 @@ PluginPinConnectorView::PluginPinConnectorView(PluginPinConnector* model, QWidge
 	, m_inView{new MatrixView{this, model->in(), true}}
 	, m_outView{new MatrixView{this, model->out(), false}}
 {
+	assert(model != nullptr);
 	connect(model, &PluginPinConnector::propertiesChanged, this, &PluginPinConnectorView::updateGeometry);
 
-	setWindowTitle(tr("Plugin Pin Connector"));
+	const Model* parentModel = model->parentModel();
+	assert(parentModel != nullptr);
```

Comment:
If `parentModel` is ever null, it would be a bug, so that's why I'm using `assert`. Setting the window title to something else in that case would just hide the bug. None of the `assert`s introduced by this PR should ever fail assuming there are no design errors or logic errors in this PR.